src/views/HomeView.vue

```diff
@@ -65,24 +75,40 @@ const quizStore = useQuizStore()
 const role = ref<Role | null>(null)
 const localSelectedRole = ref<string | null>(null)
 
+const roleSectionRef = ref<HTMLElement | null>(null)
+const knowledgeSectionRef = ref<HTMLElement | null>(null)
+const resultSectionRef = ref<HTMLElement | null>(null)
+
 onMounted(async () => {
   await rolesStore.fetchRoles()
   if (rolesStore.roles.length > 0) {
     localSelectedRole.value = ''
-    handleRoleChange(rolesStore.roles[0])
   }
 })
 
-const handleRoleChange = (selectedRole: Role) => {
-  try {
-    role.value = selectedRole
-    quizStore.initializeQuizAnswers(selectedRole.knowledges)
-    quizStore.setSelectedRole(selectedRole)
-  } catch (error) {
-    ErrorMessage('Failed to set selected role')
+const handleRoleChange = async (RoleName: string) => {
+  const selectedRole = rolesStore.roles.find((role) => role.name === RoleName)
+  if (selectedRole) {
```

Comment:
Instead of put this if here, you can check if selectedRole doesn't exist. If it's the case, the user should be aware